### ISSUE TITLE
new Aqara Wireless Remote Switch Single added as unknown Zigbee device

### ISSUE THREAD

**Issue by robinbakkenes** [TedTolboom/com.xiaomi-mi-zigbee#88]
Hi, 

I just ordered a couple of new Aqara Wireless Remote Switch Single devices. I alreay have some installed fine, but these new ones are added as unknown zigbee devices. I already rebooted Homey and the Xiaomi app a couple of times and readded them, but they are not working as expected. Perhaps the devices have a new product ID? 

![xiaomi1](https://user-images.githubusercontent.com/42275761/45869046-b0d84c00-bd87-11e8-87e0-0fa8bef50982.JPG)

**Comment by robinbakkenes** [comment]
I see the product ID for the older remote switches is different: 

![xiaomi2](https://user-images.githubusercontent.com/42275761/45883599-cbbfb600-bdb1-11e8-9452-27e69e517025.JPG)

instead of the old ID lumi.sensor_86sw1lu its now lumi.remote.b186acn01 

The device and profile ID are the same though.. Is this something that must be added to the app to be able to work again?

**Comment by Mustang1966** [comment]
Mine is also added as unknown device. 
![image](https://user-images.githubusercontent.com/43118018/45895532-b6f31a80-bdd1-11e8-91c3-d5a39e40c8d6.png)

**Comment by TedTolboom** [comment]
What is the product number on the back of the Aqara wireless switch??

**Comment by robinbakkenes** [comment]
Hi Ted,

Thanks for your reply. Hereby a photo of the backside: 
![schermafbeelding 2018-09-22 om 08 57 20](https://user-images.githubusercontent.com/42275761/45914427-88af2280-be45-11e8-8619-99db41320284.png)

WXKG03LM 
CMIIT ID: 2016DP1334
CR2032

**Comment by TedTolboom** [comment]
> The device and profile ID are the same though.. Is this something that must be added to the app to be able to work again?

Yes, that is correct... correct driver identification is based on all 4 ID's..

Since the product type number is still WXKG03LM, I would assume that the device logic is still the same...

It would help me greatly if you could confirm that the device is working based on the same driver, by following the next steps:
1. download the app from Github (https://github.com/TedTolboom/com.xiaomi-mi-zigbee)
2. add the additional product ID to line 1679 (https://github.com/TedTolboom/com.xiaomi-mi-zigbee/blob/master/app.json#L1679) by replacing:
```
"productId": [
  "lumi.sensor_86sw1lu"
 ],
```
with
```
"productId": [
  "lumi.sensor_86sw1lu",
  "lumi.remote.b186acn01"
 ],
```
3. remove the device from Homey
4. install the app through the Command Line Interface (https://community.athom.com/t/how-to-cli-install-method/198)

**Comment by robinbakkenes** [comment]
Im fairly new to this, can I test this without having to readd all my other xiaomi devices and flows?

**Comment by robinbakkenes** [comment]
I followed the steps and inserted the line. But when i want to install this new app via the command line is see this error:

D:\Homeystuff\com.xiaomi-mi-zigbee-master>athom app install
┌────────────────────────────────────────────┐
│ Hey developer, we're hiring! View our open │
│ positions at https://go.athom.com/jobs     │
└────────────────────────────────────────────┘
V Pre-processing app...
Error in `app.json`:
SyntaxError: Unexpected string in JSON at position 44963

D:\Homeystuff\com.xiaomi-mi-zigbee-master>

I didnt change anything except adding the additional line at 1679

**Comment by TedTolboom** [comment]
on line 1679 you are missing a comma at the end....
should be: `"lumi.sensor_86sw1lu",`

**Comment by TedTolboom** [comment]
Why am I not surprised...

Can you unquote (remove `//`) the debug logging (on line 12, https://github.com/TedTolboom/com.xiaomi-mi-zigbee/blob/master/drivers/sensor_86sw1lu/device.js#L12) and the printNode command (on line 15, https://github.com/TedTolboom/com.xiaomi-mi-zigbee/blob/master/drivers/sensor_86sw1lu/device.js#L15) and save the file.

Can you then run the file in debug mode `athom app run` and post the printNode of sensor_86sw1lu?

afterwards, you can stop the debug mode with CTRL-C and install the app again via `athom app install`...

**Comment by robinbakkenes** [comment]
Is this often the case with Xiaomi products? :-) 

Ill look into this as soon as possible, quite busy on personal level right now. Ill post you the logging as soon as i find the time to check it out.

**Comment by TedTolboom** [comment]
Strange thing is that they still use the same product number, while reporting different ID's from SW perspective... typically, the different ID's point towards a different implementation on SW side...

So it was a long shot... and again proven to be more difficult to resolve.
Homey's current Zigbee implementation does not provide quite much debug logging (options)...

So let's hope I can debug the issue with your support, otherwise I'll need to be able to work with the device myself (utilizing my Zigbee sniffer)...

**Comment by robinbakkenes** [comment]
Ok, ill send you the required info when i have some free time at home. Anyway, are you in the Netherlands? If so i can send you the device so you can work on it? You can send it back to me when fixed. Is that an idea?

**Comment by TedTolboom** [comment]
> Ok, ill send you the required info when i have some free time at home. Anyway, are you in the Netherlands? If so i can send you the device so you can work on it? You can send it back to me when fixed. Is that an idea?

Yes, I'm living in the southern part of the Netherlands.
Will get some feedback from my CEO that I'll need to start a warehouse to facilitate these options ;-P

**Comment by robinbakkenes** [comment]
I have run the app with debugging and prindnode on and ported the output with the -> command to a text file. Is that what you meant? the output is added:

[debuglog.txt](https://github.com/TedTolboom/com.xiaomi-mi-zigbee/files/2412456/debuglog.txt)

Or, can i find the printnode output more readable anywhere else?

**Comment by TedTolboom** [comment]
Thanks this is ok...

You’re using multiple devices with the ‘sensor_86sw1lu’ driver; are these all devices with the same ID’s or what is the mix between ID’s?

**Comment by robinbakkenes** [comment]
I now have two switches added, one who does work (an old one) and the new one who doesnt. They dont have the same ID, they are the two devices mentioned in the beginning of this  topic, lumi.sensor_86sw1lu and lumi.remote.b186acn01

**Comment by TedTolboom** [comment]
@robinbakkenes please see: https://community.athom.com/t/xiaomi-aqara-smart-home-zigbee-app-v0-4-1/156/57

**Comment by robinbakkenes** [comment]
I can but I'm on holiday right now until mid next week.  I'll contact you when I'm back! I btw also have a dual button switch which I can send

**Comment by TedTolboom** [comment]
@robinbakkenes off course enjoy your holiday first..

I would be helped if you could send me the single button version... 
the double button version I already received from another user..

**Comment by TedTolboom** [comment]
Hi Robin,

It's not possible to send a private message through Github.

Can you contact me through the [Athom community forum](https://community.athom.com) or on [Homey Slack community](https://athomcommunity.slack.com/team/tedtolboom)?

Thanks, Ted

**Comment by robinbakkenes** [comment]
on the athom community forum i dont see any option to send you a private message. The homey slack community im not a member of. 

Please contact met via mail on robinbakkenes@gmail.com

**Comment by TedTolboom** [comment]
Hi @robinbakkenes,
I’ve send you a message on the address provided; did you receive it?

**Comment by robinbakkenes** [comment]
Hey Ted,

Yeah sorry, like i said in a private message, i was unable to send the switch due to personal reasons. I will post the switch tomorrow, you' ll have it on Friday :)

**Comment by TedTolboom** [comment]
v0.4.3 solving this issue / feature request has been submitted to the app store
I will close this issue for now. Please re-open if not solved in v0.4.3
